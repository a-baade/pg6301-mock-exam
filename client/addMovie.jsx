import React, { useContext, useState } from "react";
import { MoviesApiContext } from "./moviesApiContext";

function FormInput({ label, value, onChangeValue }) {
  return (
    <div className="form-input">
      <label>
        <strong>{label}</strong>{" "}
        <input value={value} onChange={(e) => onChangeValue(e.target.value)} />
      </label>
    </div>
  );
}

export function AddNewMovie() {
  const { createMovie } = useContext(MoviesApiContext);
  const [title, setTitle] = useState("");
  const [plot, setPlot] = useState("");
  const [country, setCountry] = useState("");
  const [year, setYear] = useState("");

  async function handleSubmit(event) {
    event.preventDefault();

    createMovie({
      title,
      plot,
      year: parseInt(year),
      country,
    });
    setTitle("");
  }

  return (
    <form onSubmit={handleSubmit}>
      <h2>Add new movie</h2>

      <FormInput label="Title:" value={title} onChangeValue={setTitle} />

      <FormInput label="Plot:" value={plot} onChangeValue={setPlot} />

      <FormInput label="Country:" value={country} onChangeValue={setCountry} />

      <FormInput
        type="number"
        max="10"
        min="0"
        label="Year:"
        value={year}
        onChangeValue={setYear}
      />
      <input type="submit" value="Submit" />
    </form>
  );
}
