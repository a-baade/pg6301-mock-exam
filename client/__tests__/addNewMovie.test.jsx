import { AddNewMovie } from "../addMovie";
import React from "react";
import ReactDOM from "react-dom";
import { MemoryRouter } from "react-router-dom";

describe("add movie component", () => {
  it("shows movies form", () => {
    const element = document.createElement("div");
    ReactDOM.render(
      <MemoryRouter>
        <AddNewMovie />
      </MemoryRouter>,
      element
    );
    expect(element.innerHTML).toMatchSnapshot();
    expect(
      Array.from(element.querySelectorAll("form label strong")).map(
        (e) => e.innerHTML
      )
    ).toEqual(["Title:", "Plot:", "Country:", "Year:"]);
  });
});
