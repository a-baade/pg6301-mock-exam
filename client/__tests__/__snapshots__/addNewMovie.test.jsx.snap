// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add movie component shows movies form 1`] = `"<form><h2>Add new movie</h2><div class=\\"form-input\\"><label><strong>Title:</strong> <input value=\\"\\"></label></div><div class=\\"form-input\\"><label><strong>Plot:</strong> <input value=\\"\\"></label></div><div class=\\"form-input\\"><label><strong>Country:</strong> <input value=\\"\\"></label></div><div class=\\"form-input\\"><label><strong>Year:</strong> <input value=\\"\\"></label></div><input type=\\"submit\\" value=\\"Submit\\"></form>"`;
